src/options/outputc.h

```diff
@@ -0,0 +1,38 @@
+#include "cvc5_private_library.h"
+
+#ifndef CVC5__OPTIONS__OUTPUTC_H
+#define CVC5__OPTIONS__OUTPUTC_H
+
+#include <iostream>
+
+#include "cvc5_export.h"
+#include "base/output.h"
+#include "options/base_options.h"
+
+namespace cvc5 {
+
+class OutputC
+{
+ public:
+  explicit OutputC(std::ostream* os) : d_os(os) {}
+
+  Cvc5ostream operator()(const options::OutputTag tag) const;
+
+  bool isOn(const options::OutputTag tag) const;
+
+ private:
+  std::ostream* d_os;
+
+}; /* class OutputC */
+
+extern OutputC OutputChannel CVC5_EXPORT;
+
+#ifdef CVC5_MUZZLE
+#define Output ::cvc5::__cvc5_true() ? ::cvc5::nullStream : ::cvc5::OutputChannel
```

Comment:
Why can't we use 
```
#define Output true ? ::cvc5::nullStream : ::cvc5::OutputChannel
```
or even
```
#define Output ::cvc5::nullStream
```
here?

Reply:
Here's what the comment for `__cvc5_true` says:
```
// Disallow e.g. !Debug("foo").isOn() forms
// because the ! will apply before the ? .
// If a compiler error has directed you here,
// just parenthesize it e.g. !(Debug("foo").isOn())
```
Now I wonder why e.g. `Debug` is not defined as:

```
#define Debug (::cvc5::__cvc5_true() ? ::cvc5::nullStream : ::cvc5::DebugChannel)
```

and as you said, if we'd just just `#define Debug ::cvc5::nullStream`, there would be no need for cvc5_true at all...

Reply:
Oh OK I think I know why...

```
Debug("foo") << "bar";
```

expands to

```
Debug ::cvc5::__cvc5_true() ? ::cvc5::nullStream : ::cvc5::DebugChannel("foo") << "bar"
```

for muzzled builds `::cvc5::DebugChannel("foo") << "bar"` will get optimized away.